addons/account/models/account_move_line.py

```diff
@@ -817,7 +817,7 @@ def _compute_product_uom_id(self):
     def _compute_quantity(self):
         for line in self:
             if line.display_type == 'product':
-                line.quantity = line.quantity if line.quantity else 1
+                line.quantity = line.quantity if line.quantity is False else 1
             else:
                 line.quantity = False
```

Comment:
So just two things wih this line, first is easy, use `not [var]` instead of `[var] is False`.

```suggestion
        for line in self:
            if line.display_type == 'product':
                line.quantity = line.quantity if not line.quantity else 1
            else:
                line.quantity = False
```

Second thing is that this diff is really weird. For products, we used to use line.quantity's value, now we only check whether it's Falsy or not. Old behavior was:
0 => 1
1 => 1
2 => 2
Now we have:
0 => 0
1 => 1
2 => 1
I thought we would want to have the following:
0 => 0
1 => 1
2 => 2
At least that's what i find the most logical. In that case you would only need somehting doing this:

```suggestion
        for line in self:
            if line.display_type == 'product':
                line.quantity = line.quantity
            else:
                line.quantity = False
```

Reply:
You're right, this works for me :+1:

Reply:
scratch that, I think a context may actually be needed